### DISCUSSION TITLE
Help with a SK6812 RGBW setup

### DISCUSSION THREAD

**Discussion by jt7005** [awawa-dev/HyperHDR#242] [Q&A] [answered]
Hi,  I a m running HyperHDR on raspberry pi 3, and using  HyperSPI and seem to work fine, my question is  HyperSPI  the best option for me or should I be trying  to get HyperSerialESP32 or HyperSerialEsp8266 working.
I want to make the most of my 260 SK6812 RGBW leds.

**Comment by awawa-dev** [comment, marked as answer]
Hi
Yes, it's best option for your setup. For 260 leds it doesn't matter if ESP32 or ESP8266 is used for HyperSPI: for much larger setups ESP32 is recommended for HyperSPI because of larger SPI communication frame. 

HyperSerialESP32/HyperSerialEsp8266 should be rather used for PC/mac (lack of SPI interface) or when the distance between Rpi and ESP is quite long (USB connection can be used for much longer distance than SPI). And HyperSerialESP32/HyperSerialEsp8266 require special serial chip on ESP board that is capable to achieve 2Mb, but it's rather a shopping list problem than design ;)

**Reply by jt7005** [reply]
Thank you, it now makes sense to me
